### DISCUSSION TITLE
Running medusa in yarn v3 monorepo

### DISCUSSION THREAD

**Discussion by genox** [medusajs/medusa#5244] [Q&A]
Hi all,

I am trying to move out custom plugins into their own packages to share them between projects, not necessarily wanting to publish them to NPM.

So I thought this should be very well possible using yarn workspaces and set up a project where I have two packages, one consisting of the previous medusa backend and a plugin scaffold.

But there are numerous issues with dependencies, no matter which variant of resolver (via yarnrc) I try.

Is there anybody that managed to build/use their project this way?

Or is there another way to use medusa in a mono repo setup with local packages as dependencies?

**Comment by buffcode** [comment]
Same problem here. I managed setup the monorepo with PNPM and `shamefully-hoist`. Alternatively you can use `yalc` during development to link the build plugins (worked with Yarn Classic).

There is too much dependency magic going on in Medusa IMHO that breaks package managers.
